src/puppet.ts

```diff
@@ -689,7 +690,27 @@ export abstract class Puppet extends EventEmitter {
    * Friendship
    *
    */
-  public abstract async friendshipSearch (searchId: string) : Promise<void>
+  // return contact id by phone number
+  public abstract async friendshipSearchPhone (phone: string) : Promise<string | null>
+  // return contact id by weixin id
+  public abstract async friendshipSearchWeixin (weixin: string) : Promise<string | null>
+
+  public async friendshipSearch(searchQueryFilter: FriendshipSearchQueryFilter): Promise<string | null> {
+    log.verbose('Puppet', 'friendshipSearch("%s")', JSON.stringify(searchQueryFilter))
+
+    if (Object.keys(searchQueryFilter).length !== 1) {
+      throw new Error('searchQueryFilter should only include one key for query!')
+    }
+
+    if (searchQueryFilter.phone) {
+      return this.friendshipSearchPhone(searchQueryFilter.phone)
+    } else if (searchQueryFilter.weixin) {
+      return this.friendshipSearchWeixin(searchQueryFilter.weixin)
```

Comment:
These two methods named `friendshipSearchPhone `, `friendshipSearchPhone` will call the same API in base level. So I want to make sure that do we need to named two methods which do the same things in padplus?